packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+
+Now that `@hint/utils-test-server` has been updated, what happens to all the other packages? The dependency will be updated but no version will be published because there will be only one `Upgrade` commit in the history. Those packages have a dependency such as:
+
+```json
+"@hint/utils-test-server": "^2.0.1"
+```
+
+When someone install the package from a clean cache version `2.0.2` will be downloaded among the latest `@hint/utils-create-server"` which is fully compatible with the current code.
+
+### New feature
+
+* `packageA` breaks something → `Breaking: Drastic change in the API`.
+* `packageB` updates the code to adapt to `packageA`'s new API and it can now support a new feature → `New: Shiny feature`.
+
+The release script will:
+* releases `packageA` with a `major` bump: version 2.0.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 2.0.0`. Because the commit history has `New` and `Upgrade`, `pakageB` will be bumped to 1.6.0 (`New` > `Upgrade`).
+
+**⁉️ NOTE**: When looking at the commit history it might be surprising to see something like the following (from newer to older):
+
+```
+Upgrade: Bump packageA to 2.0.0
+New: This shiny new feature
+```
+
+instead of the reverse. Because of the way that monorepos work this is actually expected. Also this will be "hidden" from the changelog as `Upgrade:` and `Chore:` are not outputed in the process.
+
+### Breaking change
+
+* `packageA` breaks something → `Breaking: Drastic change in the API`.
+* `packageB` changes also needs to change completely to make sure how everything works → `Breaking: Drastic change in the API`.
+
+The release script will:
+* releases `packageA` with a `major` bump: version 2.0.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 2.0.0`. Because the commit history has `Breaking` and `Upgrade`, `packageB` will be bumped to 2.0.0 (`Breaking` > `Upgrade`).
+
+E.g.: If `hint` changes the expectations of the resources should look like, updating will be a `Breaking`.
+
+**⁉️ NOTE**: `packageA` commit could be anything, the important part is how the changes are handled in `packageB` and how they are commited. It is very tempting to do all the changes in one single commit but this could have very bad consequences (e.g.: bumping to a `major` all packages when it should be a `patch`).
+
+## Current release process
+
+The previous is implemented (more or less accurate) during the release process. To start a new release a use needs to run `npm run release` that executes `scripts/release.ts`. The gist is:
+
+1. Calculate what packages need to be published. To do this searches the latest tag for each package and checks if there have been any commits since then that affect it.
+1. Sort the dependencies by less dependants, i.e. if `packageA` depends on `packageB` it will work on `packageB` first.
+1. For each package:
+   * Bump the `version` in the `package.json` of the package usint the related commit message(s) and update `CHANGELOG.md`
```

Comment:
using

Reply:
Numbers shouldn't increase, this is done automatically when transforming to HTML